### ISSUE TITLE
arrow-body-style breaks when returning one line object

### ISSUE THREAD

**Issue by apexskier** [eslint/eslint#6925]
**What version of ESLint are you using?**
3.2.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```
module.exports = {
    parser: 'babel-eslint',
    rules: {
        'arrow-body-style': 2
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
```js
const obj = {};
const arr = [];
const test = arr.map(it => {
    return { test: it };
});
```

**What did you expect to happen?**
This should work with no errors, since `const test2 = arr.map(it => { test: it });` is a syntax error.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ ./node_modules/.bin/eslint temp.js 

/Users/cameronlittle/remitly-vagrant/html/narwhal/temp.js
  3:28  error  Unexpected block statement surrounding arrow body  arrow-body-style

✖ 1 problem (1 error, 0 warnings)
```

Eslint doesn't handle returning objects in arrow functions well. It wants me to put the returned object on the same line as the arrow function instead of in a block. Since the object I'm returning has it's own curly braces, that's a syntax error.

**Comment by platinumazure** [comment]
You can do a concise arrow body if you surround the expression in parentheses:

```js
const obj = {};
const arr = [];
const test = arr.map(it => ({
    return { test: it };
}));
```

**Comment by apexskier** [comment]
That works!

**Comment by bhargavi3** [comment]
const WrapNavLink = ({organization, title}) => {
  return (
    <NavLink
      activeClassName="active"
      tag={RNavLink}
      to={organization}>
      {title}
    </NavLink>
  );
};

How to solve this arrowbody error can you guys please help
